crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/generated_acir.rs

```diff
@@ -263,15 +282,26 @@ impl GeneratedAcir {
     ///
     /// (1) is because an [`Expression`] can hold at most a degree-2 univariate polynomial
     /// which is what you get when you multiply two degree-1 univariate polynomials.
-    pub(crate) fn mul_with_witness(&mut self, lhs: &Expression, rhs: &Expression) -> Expression {
+    pub(crate) fn mul_with_witness(
```

Comment:
I think adding errors to this function is overkill. The `expects` are tied to preconditions which are asserted immediately prior to them rather than relying on external data so these are correct idiomatic usage of `expect`.

This made me notice that we have an inconsistency in the logic and error message here though so have opened https://github.com/noir-lang/noir/pull/2078